x-pack/plugins/fleet/server/services/agents/update_agent_tags_action_runner.ts

```diff
@@ -108,7 +110,13 @@ export async function updateTagsBatch(
     });
     const hostedIds = hostedPolicies.items.map((item) => item.id);
 
-    query = getElasticsearchQuery(options.kuery, false, false, hostedIds);
+    const extraFilters = [];
+    if (options.tagsToAdd.length === 1 && options.tagsToRemove.length === 0) {
```

Comment:
Why do we only add a filter if `options.tagsToAdd.length === 1` could we not add more filters if `options.tagsToAdd.length > 1` ?

Reply:
Explained in the description, we could add more filters, though it would be more complex, something like `NOT tags:newTag1 OR NOT tags:newTag2 OR tags:existingTag1 OR tags:existingTag2`